hydra-redis/querying_mechanism.py

```diff
@@ -0,0 +1,463 @@
+        """
+        union = 0
+        faceted_list = []
+        while(1):
+            if query.count(" ") > 1:
+                key, value, query = query.split(" ", 2)
+                faceted_list.append(self.faceted_key(key, value))
+            else:
+                key, value = query.split(" ")
+                query = ""
+                faceted_list.append(self.faceted_key(key, value))
+            if len(query) > 0:
+                operation, query = query.split(" ", 1)
+                if operation == "or":
+                    union = 1
+
+            else:
+                break
+        if union == 1:
+            get_data = self.connection.sunion(*faceted_list)
+            print(get_data)
+        else:
+            get_data = self.connection.sinter(*faceted_list)
+            print(get_data)
+
+
+class QueryFacades:
+    """
+    It is used for call the above defined functions based on given query.
+    """
+
+    def __init__(self, api_doc, url):
+
+        self.endpoint_query = EndpointQuery()
+        self.api_doc = api_doc
+        self.url = url
+        self.properties = PropertiesQuery()
+        self.compare = CompareProperties()
+
+    def initialize(self):
+        """
+        Initialize is used to initialize the graph for given url.
+        """
+        print("just initialize")
+
+        graph.main(self.url, self.api_doc)
+
+    def user_query(self, query):
+        """
+        It calls function based on queries type.
+        """
+        query = query.replace("show ", "")
+        if query == "endpoints":
+            self.endpoint_query.get_allEndpoints(query)
+        elif query == "classEndpoints":
+            self.endpoint_query.get_classEndpoints(query)
+        elif query == "collectionEndpoints":
+            self.endpoint_query.get_collectionEndpoints(query)
+        elif "members" in query:
+            self.members = CollectionmembersQuery(self.api_doc, self.url)
+            self.members.get_members(query)
+        elif "objects" in query:
+            self.properties.get_members_properties(query)
+        elif "object" in query:
+            self.properties.get_object_property(query)
+        elif "Collection" in query:
+            self.properties.get_collection_properties(query)
+        elif "class" in query and "property_value" in query:
+            self.class_property = ClassPropertiesValue(self.api_doc, self.url)
+            self.class_property.get_property_value(query)
+        elif "class" in query:
+            self.properties.get_classes_properties(query)
+        else:
+            self.compare.object_property_comparison_list(query)
+
+
+def main():
+    """
+    Take URL as an input and make graph using initilize function.
+    Querying still user wants or still user enters the exit.
+    """
+    url = input("url>>>")
+    handle_data = HandleData()
+    apidoc = handle_data.load_data(url + "/vocab")
+    while (1):
```

Comment:
Use `while(True)` instead. Also, you should display what errors occurred in APIDoc loading failed.

Reply:
Okay.
Yeah, we are displaying the error in HandleData (line 26-30).